code/modules/hydroponics/seeds.dm

```diff
@@ -317,16 +320,32 @@
 	return
 
 /obj/item/seeds/attackby(obj/item/O, mob/user, params)
-	if (istype(O, /obj/item/plant_analyzer))
+	if(istype(O, /obj/item/plant_analyzer))
 		to_chat(user, "<span class='info'>*---------*\n This is \a <span class='name'>[src]</span>.</span>")
 		var/text = get_analyzer_text()
 		if(text)
 			to_chat(user, "<span class='notice'>[text]</span>")
 
 		return
+	if(istype(O, /obj/item/pen))
+		var/V = sanitize(stripped_input(user, "Choose variant name:", "Plant Variant Naming", variant, MAX_MESSAGE_LEN))
+		if(!Adjacent(user)) //prevent remote naming of seeds
+			return
+		if(V == "" && variant != "")
+			user.show_message("<span class='info'>You remove the variant designation from the [plantname].</span>")
+		else if(V != "")
+			user.show_message("<span class='info'>You designate the [plantname] as the [V] variant.</span>")
+		variant = V
+		apply_variant_name()
 	..() // Fallthrough to item/attackby() so that bags can pick seeds up
 
-
+/obj/item/seeds/proc/apply_variant_name()
+	var/P = findtext(name, "\[")
+	var/V = (variant == "") ? "" : (" \[" + variant + "]") // If we have a non-empty variant add it to the name
+	if (P != 0)
+		name = copytext(name, 1, P - 1) + V // If there was already a variant in the name remove it
```

Comment:
Consider the `initial(varname)` proc to get the initial name of a datum, would be better than text manipulation

Reply:
Wouldn't that delete the 'experimental' from modified seeds?

Reply:
Good point, this is fine then